src/routes/authentication.js

```diff
@@ -137,6 +139,17 @@ async function authenticationCallback(context, options, request, response, next)
   }
 }
 
+/**
+ * @param {import('../context/init').Context} context
+ * @param {import('express').Request} request
+ * @param {import('express').Response} response
+ */
+async function logout(context, request, response) {
+  const deletedToken = context.tokenService.deleteToken();
+  const token = request.headers.cookie.split('=')[1];
```

Comment:
There is an error with this code when there are no cookies at all (when the agent is on localhost)

```UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'split' of undefined
    at _callee3$ (/Users/gg/repositories/forest-express/dist/routes/authentication.js:225:44)
    at tryCatch (/Users/gg/repositories/forest-express/node_modules/regenerator-runtime/runtime.js:45:40)
    at Generator.invoke [as _invoke] (/Users/gg/repositories/forest-express/node_modules/regenerator-runtime/runtime.js:274:22)
    at Generator.prototype.<computed> [as next] (/Users/gg/repositories/forest-express/node_modules/regenerator-runtime/runtime.js:97:21)
    at asyncGeneratorStep (/Users/gg/repositories/forest-express/node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
    at _next (/Users/gg/repositories/forest-express/node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
    at /Users/gg/repositories/forest-express/node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7
    at new Promise (<anonymous>)
    at /Users/gg/repositories/forest-express/node_modules/@babel/runtime/helpers/asyncToGenerator.js:21:12
    at _logout (/Users/gg/repositories/forest-express/dist/routes/authentication.js:236:18)
    at logout (/Users/gg/repositories/forest-express/dist/routes/authentication.js:205:18)
    at Layer.handle [as handle_request] (/Users/gg/repositories/forest-express/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/gg/repositories/forest-express/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/Users/gg/repositories/forest-express/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/Users/gg/repositories/forest-express/node_modules/express/lib/router/layer.js:95:5)
    at /Users/gg/repositories/forest-express/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/Users/gg/repositories/forest-express/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/gg/repositories/forest-express/node_modules/express/lib/router/index.js:275:10)
    at /Users/gg/repositories/forest-express/node_modules/express-jwt/lib/index.js:131:7
    at /Users/gg/repositories/forest-express/node_modules/async/lib/async.js:52:16
    at Immediate.<anonymous> (/Users/gg/repositories/forest-express/node_modules/async/lib/async.js:1206:34)
    at processImmediate (internal/timers.js:439:21)
```


+ I think that your code has a problem when there are more than one cookie set.

Reply:
A better approach would be to extract the code from `src/index.js` and reuse it in both places.